src/lib.rs

```diff
@@ -638,16 +845,19 @@ impl<T> core::future::Future for Receiver<T> {
     type Output = Result<T, RecvError>;
 
     fn poll(self: Pin<&mut Self>, cx: &mut task::Context<'_>) -> Poll<Self::Output> {
+        // TODO: fix this. all of it.
+
+        // SAFETY: The channel will not be freed while this method is still running.
         let channel = unsafe { self.channel_ptr.as_ref() };
 
-        match channel.state.load(SeqCst) {
+        match channel.state.load(Acquire) {
             // The sender is alive but has not sent anything yet.
             EMPTY => unsafe { channel.write_async_waker(cx) },
             // We were polled again while waiting for the sender. Replace the waker with the new one.
             RECEIVING => {
                 match channel
                     .state
-                    .compare_exchange(RECEIVING, EMPTY, SeqCst, SeqCst)
+                    .compare_exchange(RECEIVING, EMPTY, Relaxed, Relaxed)
```

Comment:
Can we get a motivation for this ordering please? Or revert it to something stronger. Even if we want to move all the `Future` stuff out to its own type. I need to be able to release a `0.1.4` with all these fixes and the same API first. So we need to make sure the `Future` implementation is sound even before doing the bigger refactor.